src/cloudevents/core/formats/json.py

```diff
@@ -0,0 +1,74 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+import base64
+import re
+from datetime import datetime
+from json import JSONEncoder, dumps
+from typing import Any, Final, Pattern, Union
+
+from cloudevents.core.base import BaseCloudEvent
+from cloudevents.core.formats.base import Format
+
+
+class _JSONEncoderWithDatetime(JSONEncoder):
+    """
+    Custom JSON encoder to handle datetime objects in the format required by the CloudEvents spec.
+    """
+
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, datetime):
+            dt = obj.isoformat()
+            # 'Z' denotes a UTC offset of 00:00 see
+            # https://www.rfc-editor.org/rfc/rfc3339#section-2
+            if dt.endswith("+00:00"):
+                dt = dt.removesuffix("+00:00") + "Z"
+            return dt
+
+        return super().default(obj)
+
+
+class JSONFormat(Format):
```

Comment:
how do you envision usage of this class by the end users?

Reply:
Technically, there should be no need for this class to be used often, unless someone knows what they are doing. 
Since for each binding, we would have to implement both structured and binary modes, this format class would be useful when using structured mode, since the entire CloudEvent class is serialized into JSON/Avro and send as a single message. This format would be injected into that "structured mode CE writer" implementation.
 
I can't see this class being used for binary mode tho, correct me if I'm wrong please. But given that attributes are sent as message metadata, it can only be JSON format by default, no? 

Anyway, I think it's worth having a way for having `CE -> bytes` and `bytes -> CE` for each supported format as per the specs and let engineers do manipulations as they see fit.